cmd/kueuectl/app/list/list_pods_test.go

```diff
@@ -0,0 +1,139 @@
+limitations under the License.
+*/
+
+package list
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestPodOptions_listPods(t *testing.T) {
+	testStartTime := time.Now()
+
+	type fields struct {
+		PrintFlags             *genericclioptions.PrintFlags
+		AllNamespaces          bool
+		Namespace              string
+		LabelSelector          string
+		FieldSelector          string
+		UserSpecifiedForObject string
+		ForObject              objectRef
+	}
+	tests := []struct {
+		name    string
+		pods    []runtime.Object
+		fields  fields
+		wantOut []string
+		wantErr error
+	}{
+		{
+			name: "list pods for batch/job type",
+			pods: []runtime.Object{
+				&corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "valid-pod-1",
+						Namespace: "default",
+						CreationTimestamp: metav1.Time{
+							Time: testStartTime.Add(-time.Hour).Truncate(time.Second),
+						},
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "batch/v1",
+								Kind:       "Job",
+								Name:       "test-job",
+							},
+						},
+					},
+					Status: corev1.PodStatus{
+						Phase: "RUNNING",
+					},
+				}, &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "valid-pod-2",
+						Namespace: "default",
+						CreationTimestamp: metav1.Time{
+							Time: testStartTime.Add(-time.Hour).Truncate(time.Second),
+						},
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "batch/v1",
+								Kind:       "Job",
+								Name:       "test-job",
+							},
+						},
+					},
+					Status: corev1.PodStatus{
+						Phase: "COMPLETED",
+					},
+				},
+			},
+			fields: fields{
+				Namespace:              "default",
+				UserSpecifiedForObject: "job/test-job",
+			},
+			wantOut: []string{
+				"NAME          STATUS      AGE",
+				"valid-pod-1   RUNNING     60m",
+				"valid-pod-2   COMPLETED   60m",
+				"",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClientset := fake.NewSimpleClientset(tt.pods...)
+			s, _, out, _ := genericiooptions.NewTestIOStreams()
+
+			o := &PodOptions{
+				PrintFlags:             tt.fields.PrintFlags,
+				AllNamespaces:          tt.fields.AllNamespaces,
+				Namespace:              tt.fields.Namespace,
+				LabelSelector:          tt.fields.LabelSelector,
+				FieldSelector:          tt.fields.FieldSelector,
+				UserSpecifiedForObject: tt.fields.UserSpecifiedForObject,
+				Clientset:              fakeClientset,
+				PrintObj:               printPodTable,
+				IOStreams:              s,
+			}
+
+			gotErr := o.parseForObject()
```

Comment:
I think it's better to test overall for command like [here](https://github.com/kubernetes-sigs/kueue/blob/main/cmd/kueuectl/app/list/list_workload_test.go#L570). Than we will cover more cases.

Reply:
Oops just saw your comments, sure will work on it.

Reply:
@mbobrovskyi -- I think I tried to use [that](https://github.com/kubernetes-sigs/kueue/blob/main/cmd/kueuectl/app/list/list_workload_test.go#L567) but I was getting -- 
```
panic: no kind is registered for the type v1.Pod in scheme "pkg/runtime/scheme.go:100" [recovered]
```

Can't able to use `"k8s.io/client-go/kubernetes/fake"`
```
Cannot use 'clientset' (type *Clientset) as the type versioned.Interface Type does not implement 'versioned.Interface' as some methods are missing: KueueV1alpha1() kueuev1alpha1.KueueV1alpha1Interface KueueV1beta1() kueuev1beta1.KueueV1beta1Interface VisibilityV1alpha1() visibilityv1alpha1.VisibilityV1alpha1Interface
```

Reply:
I think it's because you're trying to implement kueue go-client. You need to make SimpleClientset from k8s go-client. "k8s.io/client-go/kubernetes/fake"

Reply:
Yeah but I'm using the K8s sample clientset of type [k8s.interface](https://github.com/kubernetes-sigs/kueue/pull/2280/files#diff-837b77b83169ae5075d69c513a71b7dec4f7f2db26066367a87e79b17cc21b04R59) but the test clientgetter expects the Clientset in [versioned.Interface](https://github.com/kubernetes-sigs/kueue/blob/main/cmd/kueuectl/app/testing/fake.go#L16) type.

Reply:
Ah, got it. You need to make separate clientset for k8s.

Reply:
So we want to [TestClientGetter](https://github.com/kubernetes-sigs/kueue/blob/main/cmd/kueuectl/app/testing/fake.go#L16) to support the k8s.clientset? Because that's where we are passing to the `NewPodCmd`

Reply:
For sure. Need to implement [this](https://github.com/kubernetes-sigs/kueue/blob/df5517928d5a4adaac8c56e7c6429bf53713e59d/cmd/kueuectl/app/util/client_getter.go#L31) on `TestClientGetter`.